### ISSUE TITLE
[Feature Request]: Show buffer line in all (vertical) splits?

### ISSUE THREAD

**Issue by mrjones2014** [akinsho/bufferline.nvim#348]
Is something like this edited image possible? To be clear, I took 2 separate screenshots and edited them together to get this image:

![CleanShot-2022-03-29-at-13 04 56](https://user-images.githubusercontent.com/8648891/160666863-e404368a-bab4-42b7-ab10-438449357240.png)

Note that it shows the active buffer for each window, separately.

Not sure if this is possible or feasible but I'd love to see it.

**Comment by akinsho** [comment]
@mrjones2014 so this is a duplicate of a very recent feature request #345. I first saw this behaviour in Emacs' centaur tabs, which originally inspired this plugin. I've personally since come off of it for a few reasons.

Firstly it's useful in `vscode` or `emacs` because both of those have the concept of `editor groups` so rather than windows it's the files are actually specifically related to each section, whereas in vim it would literally be 20 buffers open for example and 2 splits, what should happen in that case how do all the other buffers arrange themselves?
How does it behave if there is a horizontal split on top of two vertical ones?

Even if they could be arranged sensibly, why would that be useful?
At this point for nvim it seems more like eye candy than anything that would actually improve the editing experience.

Happy to hear a detailed explanation on how it improves the editing experience, though, since I'd only personally consider this being added if it was something that a significant amount of bufferline users might want to use 🤷🏿‍♂️ or at least I could explain the utility to someone else.

Re. implementation, I guess it would look something like the logic for offsets slightly mixed with adding in a component to the tabline that was essentially just x amount of spaces as to bridge the gap between the last buffer and the window start of the next window. It shouldn't be massively complex since a lot of the behaviour is partially implemented but I think the utility is the biggest blocker.

**Comment by akinsho** [comment]
I guess if the only thing the bufferline was showing was the buffers open in the visible windows then this mode might make sense, but it still leaves the question of what to do with non-top level splits e.g. a lower horizontal split or

**Comment by mrjones2014** [comment]
Definitely understand the argument against something like this. The only real argument I have _for_ it is now that we can have a global status line (in a recent PR merged to Neovim), I'm looking for a different way to tell which file is open in which window (which, let me know if you have any alternative suggestions to solve this problem). I really like the global status line but this is one issue it causes.

>20 buffers open for example and 2 splits, what should happen in that case how do all the other buffers arrange themselves?

I'd think they'd just collapse behind each window to the right of it, if that makes sense.

> horizontal

Yeah, no idea how horizontal splits would be handled in this scenario.

If you feel it's not worth it, feel free to close the issue and I'll explore alternate solutions for my issue (just an indicator of what file is open in what window).

**Comment by mrjones2014** [comment]
Ah!!! I remember seeing that PR but couldn't find it, thanks for the link!

Maybe an alternate solution would be to allow the buffer line to show in the status line instead of tab line?

I just experimented with doing that with Lualine for example. See screenshot. Top bar is showing status in tab line, bottom bar is showing buffer list in per-window status line, with the active buffer correctly highlighted.

![CleanShot-2022-03-29-at-14 40 40](https://user-images.githubusercontent.com/8648891/160682279-d68a8011-f4b4-4ea0-a381-4581e3a20f8d.png)

**Comment by mrjones2014** [comment]
Yup, that all sounds reasonable. I'll wait for the winbar thing to be merged and make a new plugin for that.

**Comment by akinsho** [comment]
@mrjones2014 I don't think once `winbar` is merged you'll need a plugin for that, unless I'm still missing something about what you mean, which is possible

**Comment by mrjones2014** [comment]
Well I think that PR is just the APIs to set the `winbar` contents. You'll still need some code or a plugin to put the buffer list as the content of the `winbar`, unless I'm misunderstanding something.

**Comment by akinsho** [comment]
Ah that makes sense

**Comment by akinsho** [comment]
@mrjones2014 btw someone implemented this in https://github.com/b0o/incline.nvim which I've been using and is quite nice

**Comment by mrjones2014** [comment]
Neat, do you have this integrated with bufferline some way?

**Comment by akinsho** [comment]
It doesn't really need to be integrated, it just shows your current buffer name in a floating window out of the way in the corner of each window
